Add Runtime Virtual Texture support to 3D Tilesets

wickerman123 opened the pull request:
Drawing 3D Tilesets to Runtime Virtual Textures has been requested a couple times on the forums:
https://community.cesium.com/t/draw-cesium3dtiles-to-virtual-texture/17604
https://community.cesium.com/t/projecting-cesium-imagery-onto-static-mesh/21806
https://community.cesium.com/t/masking-terrain-as-per-meshes/23889

This PR simply exposes relevant Virtual Textures settings on the 3DTileset actor so that the CesiumGltfComponent can pass these arguments when constructing the StaticMeshes. The properties LodBias, CullMips and MinCoverage were omitted as in my testing, they expectedly didn't have any effect - presumably because of how Lods and Streaming are handled by the tileset.

![image](https://github.com/user-attachments/assets/2bb54fec-4d09-4fac-ad97-b790a519fd18)

Included with these changes are a Layer Blend for Angle Masking and a Material Layer for overlaying an RVT's BaseColor.
![image](https://github.com/user-attachments/assets/61ad5e63-fb25-449c-9d7e-a30a706b0867)
An example usecase is taking imagery from Cesium World Terrain and overlaying it OSM Building's Rooves. While the RasterOverlayComponent can already do this, using an RVT allows for a consistent and far higher texel density when close to the surface.

_Raster Overlay:_
![RasterOverlay](https://github.com/user-attachments/assets/51a00b21-208b-4923-8591-3f8c9c1dfe08)
_RVT Overlay_
![RVTOverlay](https://github.com/user-attachments/assets/52a2cf71-f5e9-4285-a1d3-477e397f0689)

The CesiumBaseMaterial had to be modified to support this as unfortunately, writing to an RVT can't be added to a material via Material Layers.
![image](https://github.com/user-attachments/assets/38e4f49e-dfb2-4e56-aeea-84dedefe7ddb)



I have also worked on a demo scene for addition to the samples project at a later date:
![image](https://github.com/user-attachments/assets/f7aad39b-ba14-46dd-8f8e-49059057936d)

azrogers commented:
Thanks for the PR @wickerman123! Getting this feature in would be great, and thank you so much for taking on the work yourself. Just a few things to tackle:

- Please update `CHANGES.md` with a short description of your additions here, so we can include it in the changelog.
- Can you format the code? This is that "Quick Checks" failure on CI above. It's pretty simple to do - just run `npm install` and `npm run format` in the plugin directory.
- I notice you need to manually refresh the tileset to apply the `VirtualTextureRenderPassType` property. Is there a good reason for this manual refresh? If not, can you add this property to the list of checks in `PostEditChangeProperty` so it can be refreshed automatically when the property is changed?

azrogers commented:
Oh, and a very important point I shouldn't forget: can you fill out the Contributor License Agreement form? You can find links to the forms for individuals and corporations here: https://github.com/CesiumGS/community/tree/main/CLAs. We'll need to get this filled out to be able to merge your code. Thank you again!

wickerman123 commented:
> Thanks for the PR @wickerman123! Getting this feature in would be great, and thank you so much for taking on the work yourself. Just a few things to tackle:
> 
> * Please update `CHANGES.md` with a short description of your additions here, so we can include it in the changelog.
> * Can you format the code? This is that "Quick Checks" failure on CI. It's pretty simple to do - just run `npm install` and `npm run format` in the plugin directory.
> * I notice you need to manually refresh the tileset to apply the `VirtualTextureRenderPassType` property. Is there a good reason for this manual refresh? If not, can you add this property to the list of checks in `PostEditChangeProperty` so it can be refreshed automatically when the property is changed?

Should be all formatted now and I updated Changes.md.

The manual refreshing was because I didn't think to check if PostEditChangeProperty was already implemented (I should have realised, given the actors behaviour) and didn't want to implement it myself because of the potential cost it incurs! Being overly cautious with my first PR😅. I've added the new properties to the list and removed mention of refreshing from the tooltip.

wickerman123 commented:
> Oh, and a very important point I shouldn't forget: can you fill out the Contributor License Agreement form? You can find links to the forms for individuals and corporations here: https://github.com/CesiumGS/community/tree/main/CLAs. We'll need to get this filled out to be able to merge your code. Thank you again!

I've signed and sent the Individual form. 🙂

wickerman123 commented:
@azrogers Hey, just to check, is there something I need to do on my side to re-trigger the automated checks or do I just need to wait for a Cesium employee to trigger it?

azrogers commented:
@wickerman123 Someone with commit access to the repo needs to do it - which I've just done! Thank you for making the changes and signing the CLA - I will merge this if everything comes up green.

azrogers commented:
Thanks for the additions @wickerman123!
